### ISSUE TITLE
Provide link to parent category

### ISSUE THREAD

**Issue by aturon** [rust-lang/crates.io#1093]
Subcategories don't provide a way to get back to their parents. E.g., at https://crates.io/categories/development-tools::debugging, I expected "Development tools" to be a link.

**Comment by erewok** [comment]
I was looking at this and another issue ([#721](https://github.com/rust-lang/crates.io/issues/721)) and I had a suggestion for organizing the `categories` table that I think would make these things easier. 

Because the categories are hierarchical, it might be more ergonomic if the endpoints were able to rely more on the database to organize the data hierarchically.

I noticed there's a few places where [the code building SQL queries like this](https://github.com/rust-lang/crates.io/blob/726906f6e2196a5c90f0d7dda645f2ff719cf231/src/category.rs#L137) to look up parent categories:

```
split_part(c.slug, '::', 1) = c.slug
```

We could implement more proper trees in the database by utilizing either a nullable `parent_id` column on the `categories` table or adding a `path` column that is of the type `ltree` which gets updated based on the slug (automatically via triggers or in application code).

The latter appeals to me because it allows for some easier queries and because it seems more "harmless" from a migration path (the syntax does look a bit weird, perhaps, if you're not used to it). There are trade-offs, though. For instance, I think it would end up being completely invisible to application code to use an `ltree` and you'd need to hand write SQL (which the app is doing anyway to work with these categories.

I started a POC branch to work on implementing this and I believe it would work and make working with the hierarchy of categories easier, but wanted to see what the community thought. I'm new here, so happy to defer to other opinions.
